stdlib/public/core/SIMDVector.swift

```diff
@@ -1407,3 +1461,154 @@ where T: SIMD, T.Scalar: FloatingPoint {
+        CotangentVector.Scalar: BinaryFloatingPoint {
+  @inlinable
+  static func _vjpAdd(
+    lhs: Self, rhs: Self
+    ) -> (Self, (CotangentVector) -> (CotangentVector, CotangentVector)) {
+    return (lhs + rhs, { v in
+      return (v, v)
+    })
+  }
+  
+  @inlinable
+  static func _vjpSubtract(
+    lhs: Self, rhs: Self
+    ) -> (Self, (CotangentVector) -> (CotangentVector, CotangentVector)) {
+    return (lhs - rhs, { (v: CotangentVector) in
+      return (v, -v)
+    })
+  }
+}
+
+extension SIMD
+  where Self: Differentiable,
+        CotangentVector: SIMD,
+        // error: generic parameter 'Self' could not be inferred: return (lhs * rhs,...
+        Scalar : BinaryFloatingPoint,
+        // binary operator '*' cannot be applied to operands of type 'Self.CotangentVector' and 'Self'
+        Self.CotangentVector == Self {
+  @inlinable
+  static func _vjpMultiply(
+    lhs: Self, rhs: Self
+    ) -> (Self, (CotangentVector) -> (CotangentVector, CotangentVector)) {
+    return (lhs * rhs, { (v: CotangentVector) in
+      return (v * rhs, v * lhs)
+    })
+  }
+  
+  @inlinable
+  static func _vjpDivide(
+    lhs: Self, rhs: Self
+    ) -> (Self, (CotangentVector) -> (CotangentVector, CotangentVector)) {
+    return (lhs / rhs, { (v: CotangentVector) in
+      (v / rhs, -lhs / (rhs * rhs) * v)
+    })
+  }
+}
+
+extension SIMD
+  where Self : Differentiable,
+        CotangentVector : SIMD,
+        Scalar : BinaryFloatingPoint & Differentiable,
+        Scalar.CotangentVector: BinaryFloatingPoint,
+        CotangentVector.Scalar == Scalar.CotangentVector {
+  @inlinable
+  static func _vjpAdd(
+    lhs: Scalar, rhs: Self
+    ) -> (Self, (CotangentVector) -> (Scalar.CotangentVector, CotangentVector)) {
+    return (lhs + rhs, { (v: CotangentVector) in
+      return (v.sum(), v)
+    })
+  }
+
+  @inlinable
+  static func _vjpSubtract(
+    lhs: Scalar, rhs: Self
+    ) -> (Self, (CotangentVector) -> (Scalar.CotangentVector, CotangentVector)) {
+    return (lhs + rhs, { (v: CotangentVector) in
+      return (v.sum(), -v)
+    })
+  }
+
+  @inlinable
+  static func _vjpAdd(
+    lhs: Self, rhs: Scalar
+    ) -> (Self, (CotangentVector) -> (CotangentVector, Scalar.CotangentVector)) {
+    return (lhs + rhs, { (v: CotangentVector) in
+      return (v, v.sum())
+    })
+  }
+
+  @inlinable
+  static func _vjpSubtract(
+    lhs: Self, rhs: Scalar
+    ) -> (Self, (CotangentVector) -> (CotangentVector, Scalar.CotangentVector)) {
+    return (lhs + rhs, { (v: CotangentVector) in
+      return (v, -v.sum())
+    })
+  }
+}
+
+extension SIMD
+  where Self : Differentiable,
+        CotangentVector : SIMD,
+        Scalar : BinaryFloatingPoint & Differentiable,
+        Self.CotangentVector == Self,
+        Scalar.CotangentVector == Scalar {
+  @inlinable
+  static func _vjpMultiply(
+    lhs: Self, rhs: Scalar
+    ) -> (Self, (CotangentVector) -> (CotangentVector, Scalar.CotangentVector)) {
+    return (lhs * rhs, { (v: CotangentVector) in
```

Comment:
Hmm, I don't think a type signature is required for any pullback argument. If it's part of the `return` statement, it should be inferred.
```suggestion
    return (lhs * rhs, { v in
```